ios/browser/api/ads/brave_ads.mm

```diff
@@ -434,8 +434,9 @@ - (BOOL)hasViewedAdsInPreviousCycle {
 
   const auto dates = [[NSMutableArray<NSDate*> alloc] init];
   for (const auto& history_item : history_items) {
-    const auto date = [NSDate
-        dateWithTimeIntervalSince1970:history_item.created_at.ToDoubleT()];
+    const auto date =
+        [NSDate dateWithTimeIntervalSince1970:history_item.created_at
+                                                  .InSecondsFSinceUnixEpoch()];
```

Comment:
Sort of unrelated to this Chromium bump, but shouldn't we be using `Time::FromNSDate` and `Time::ToNSDate` instead of going back and forth between reference date seconds? https://source.chromium.org/chromium/chromium/src/+/main:base/time/time.h;l=730;bpv=0;bpt=1

Reply:
@kylehickinson I will check this and raise an issue if needed. Thanks for spotting this

Reply:
@tmancey Its not specific to ads, its just one of the many changes across this bump due to the removal of a a time API we were using. Since we're changing them all here anyways we may be better off just updating them in the bump to use the proper APIs cc @mkarolin

Reply:
Would rather this were a follow up. It seems like this change could be done on master even now.

Reply:
Created: https://github.com/brave/brave-browser/issues/34366